godot-ffi/src/plugins.rs

```diff
@@ -17,53 +17,44 @@
 #[macro_export]
 macro_rules! plugin_registry {
     ($vis:vis $registry:ident: $Type:ty) => {
-        $crate::paste::paste! {
-            #[used]
-            #[allow(non_upper_case_globals)]
-            #[doc(hidden)]
-            $vis static [< __godot_rust_plugin_ $registry >]:
-                std::sync::Mutex<Vec<$Type>> = std::sync::Mutex::new(Vec::new());
-        }
+        #[used]
+        #[allow(non_upper_case_globals)]
+        #[doc(hidden)]
+        $vis static $registry: std::sync::Mutex<Vec<$Type>>
+                             = std::sync::Mutex::new(Vec::new());
     };
 }
 
 #[doc(hidden)]
 #[macro_export]
-#[allow(clippy::deprecated_cfg_attr)]
-#[cfg_attr(rustfmt, rustfmt::skip)]
-// ^ skip: paste's [< >] syntax chokes fmt
+// Following rustfmt::skip is no longer needed, but there are 2 workarounds good to know, thus preserved.
+// #[allow(clippy::deprecated_cfg_attr)]
+// #[cfg_attr(rustfmt, rustfmt::skip)]
 //   cfg_attr: workaround for https://github.com/rust-lang/rust/pull/52234#issuecomment-976702997
 macro_rules! plugin_execute_pre_main_wasm {
     ($gensym:ident,) => {
         // Rust presently requires that statics with a custom `#[link_section]` must be a simple
-        // list of bytes on the wasm target (with no extra levels of indirection such as references).
+        // list of bytes on the Wasm target (with no extra levels of indirection such as references).
         //
-        // As such, instead we export a fn with a random name of predictable format to be used
-        // by the embedder.
-        $crate::paste::paste! {
-            #[no_mangle]
-            extern "C" fn [< rust_gdext_registrant_ $gensym >] () {
-                __init();
-            }
```

Comment:
Was paste not still necessary here? We need a prefix with which to search for registered functions at init time.

Reply:
If needed, we can use [`pastey`](https://lib.rs/pastey) instead.

Reply:
Thanks, good catch! This is the problem when the comments aren't expressive: "a random name of predictable format" didn't make me think that the concrete prefix mattered.

> If needed, we can use [`pastey`](https://lib.rs/pastey) instead.

I don't simply trust random crates, and this one in particular had [an incident](https://www.reddit.com/r/rust/comments/1j7fnsi/introducing_pastey_successor_of_paste/) where it just replaced all commits of the previous author. Not a great start.

I'll see if I find another solution. Maybe a proc-macro, then we can get rid of `gensym` as well.

Reply:
Thanks for checking. Let's do it ourselves then.

Reply:
Fixed in https://github.com/godot-rust/gdext/pull/1092.